pkg/webhooks/subscription/subscription.go

```diff
@@ -0,0 +1,183 @@
+package subscription
+
+import (
+	"encoding/json"
+	"net/http"
+	"sync"
+
+	responsehelper "github.com/openshift/managed-cluster-validating-webhooks/pkg/helpers"
+	"github.com/openshift/managed-cluster-validating-webhooks/pkg/webhooks/utils"
+	"k8s.io/api/admission/v1beta1"
+	admissionregv1 "k8s.io/api/admissionregistration/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+	admissionctl "sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+const (
+	WebhookName    string = "subscription-validation"
+	LoggingSubName string = "cluster-logging"
+	ESSubName      string = "elasticsearch-operator"
+	BlockedChannel string = "4.5"
+)
+
+var (
+	privilegedUsers = []string{"kube:admin", "system:admin", "system:serviceaccount:kube-system:generic-garbage-collector"}
+	adminGroups     = []string{"osd-sre-admins", "osd-sre-cluster-admins"}
+
+	log = logf.Log.WithName(WebhookName)
+
+	scope = admissionregv1.NamespacedScope
+	rules = []admissionregv1.RuleWithOperations{
+		{
+			Operations: []admissionregv1.OperationType{"UPDATE", "CREATE"},
+			Rule: admissionregv1.Rule{
+				APIGroups:   []string{"operators.coreos.com"},
+				APIVersions: []string{"*"},
+				Resources:   []string{"subscriptions"},
+				Scope:       &scope,
+			},
+		},
+	}
+)
+
+type subscriptionRequest struct {
+	Metadata struct {
+		Name      string `json:"name"`
+		Namespace string `json:"namespace"`
+	} `json:"metadata"`
+	Spec struct {
+		Channel string `json:"channel"`
+		Name    string `json:"name"`
+	} `json:"spec"`
+}
+
+// SubscriptionWebhook validates a Subscription change
+type SubscriptionWebhook struct {
+	mu sync.Mutex
+	s  runtime.Scheme
+}
+
+// TimeoutSeconds implements Webhook interface
+func (s *SubscriptionWebhook) TimeoutSeconds() int32 { return 2 }
+
+// MatchPolicy implements Webhook interface
+func (s *SubscriptionWebhook) MatchPolicy() admissionregv1.MatchPolicyType {
+	return admissionregv1.Equivalent
+}
+
+// Name implements Webhook interface
+func (s *SubscriptionWebhook) Name() string { return WebhookName }
+
+// FailurePolicy implements Webhook interface
+func (s *SubscriptionWebhook) FailurePolicy() admissionregv1.FailurePolicyType {
+	return admissionregv1.Ignore
+}
+
+// Rules implements Webhook interface
+func (s *SubscriptionWebhook) Rules() []admissionregv1.RuleWithOperations { return rules }
+
+// GetURI implements Webhook interface
+func (s *SubscriptionWebhook) GetURI() string { return "/subscription-validation" }
```

Comment:
```suggestion
func (s *SubscriptionWebhook) GetURI() string { return "/" + WebhookName }
```

Sorry for the late find. Can we do this instead please?